ca/certificates.go

```diff
@@ -288,39 +309,40 @@ func NewRootCA(cert, key []byte, certExpiry time.Duration) (RootCA, error) {
 	}
 
 	// Attempt to decrypt the current private-key with the passphrases provided
-	priv, err = helpers.ParsePrivateKeyPEMWithPassword(key, passphrase)
+	priv, err = helpers.ParsePrivateKeyPEMWithPassword(keyBytes, passphrase)
 	if err != nil {
-		priv, err = helpers.ParsePrivateKeyPEMWithPassword(key, passphrasePrev)
+		priv, err = helpers.ParsePrivateKeyPEMWithPassword(keyBytes, passphrasePrev)
 		if err != nil {
 			log.Debug("Malformed private key %v", err)
 			return RootCA{}, err
 		}
 	}
 
-	if err := ensureCertKeyMatch(parsedCA, priv.Public()); err != nil {
+	// We will always use the first certificat inside of the rootbundle as the active one
+	if err := ensureCertKeyMatch(parsedCerts[0], priv.Public()); err != nil {
```

Comment:
When it comes to these things I would rather be strict. Certificate bundles are known to have ordering: http://nginx.org/en/docs/http/configuring_https_servers.html

I'm willing to be convinced that this is worth supporting, but as long as we have good validation on the future `RootCASpec` on the correct ordering of the bundle, we should be fine.